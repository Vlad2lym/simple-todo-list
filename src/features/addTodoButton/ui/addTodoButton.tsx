import addIcon from '@/shared/images/icons/add-icon.svg';
import styles from './addTodoButton.module.scss';

interface IProps {
  className?: string;
  onClick: () => void;
}

export const test_AddTodoBtn = 'test_AddTodoBtn';

export const AddTodoButton = ({ className, onClick }: IProps) => {
  return (
    <div className={className ? `${styles.addTodoArea}} ${className}` : styles.addTodoArea}>
      <button className={styles.addTodoButton} onClick={onClick} data-testid={test_AddTodoBtn}>
        <img src={addIcon} />
      </button>
    </div>
  );
};
