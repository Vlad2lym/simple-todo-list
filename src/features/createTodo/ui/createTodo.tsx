import { useCallback, useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { Todo } from '@/entities/todo';
import { TodoForm } from '@/entities/todoForm';

interface IProps {
  onApply: (todoInfo: Todo) => void;
  onCancel: () => void;
  order: number;
}

export const CreateTodoModal = ({ onApply, onCancel, order }: IProps) => {
  const [titleTodo, setTitleTodo] = useState('');

  const onChangeTitleTodo = (value: string) => {
    setTitleTodo(value);
  };

  const onFormFinish = useCallback(() => {
    if (!titleTodo.trim()) {
      return;
    }

    const todoInfo: Todo = {
      id: uuidv4(),
      title: titleTodo,
      active: true,
      order: order,
    };

    onApply(todoInfo);
    setTitleTodo('');
    onCancel();
  }, [titleTodo, onApply, onCancel, order]);

  return (
    <>
      <TodoForm
        title="NEW NOTE"
        onCancel={onCancel}
        onApply={onFormFinish}
        inputValue={titleTodo}
        onChangeInput={onChangeTitleTodo}
      />
    </>
  );
};
