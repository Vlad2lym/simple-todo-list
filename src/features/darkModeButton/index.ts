export { DarkModeButton } from './ui/darkModeButton';
