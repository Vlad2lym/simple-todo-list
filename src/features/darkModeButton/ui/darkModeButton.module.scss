.dark-mode-button {
  width: 38px;
  height: 38px;
  background-color: var(--primary-color);
  background-image: var(--mode-button-icon);
  background-repeat: no-repeat;
  background-position: center;
  border: none;
  border-radius: 5px;
  cursor: pointer;

  &:hover {
    background-color: var(--primary-color-active);
    box-shadow: 0 0 4px 0 rgba(108, 99, 255, 1);
  }
}
