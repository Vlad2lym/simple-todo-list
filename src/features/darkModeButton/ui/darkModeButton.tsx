import styles from './darkModeButton.module.scss';

interface IProps {
  onChange: () => void;
}

export const test_DarkModeBtn = 'test_DarkModeBtn';

export const DarkModeButton = ({ onChange }: IProps) => {
  return <button className={styles.darkModeButton} onClick={() => onChange()} data-testid={test_DarkModeBtn} />;
};
