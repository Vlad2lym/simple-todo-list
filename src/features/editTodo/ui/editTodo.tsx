import { useCallback, useState } from 'react';
import { Todo } from '@/entities/todo';
import { TodoForm } from '@/entities/todoForm';

interface IProps {
  onApply: (todoInfo: Todo) => void;
  onCancel: () => void;
  editedTodo: Todo | null;
}

export const EditTodoModal = ({ onApply, onCancel, editedTodo }: IProps) => {
  const [titleTodo, setTitleTodo] = useState(editedTodo?.title ?? '');

  const onChangeTitleTodo = (value: string) => {
    setTitleTodo(value);
  };

  const onFormFinish = useCallback(() => {
    if (!titleTodo.trim() || !editedTodo) {
      return;
    }

    const todo: Todo = {
      id: editedTodo.id,
      title: titleTodo,
      active: editedTodo.active,
      order: editedTodo.order,
    };

    onApply(todo);
    onCancel();
  }, [titleTodo, onApply, onCancel, editedTodo]);

  return (
    <>
      <TodoForm
        title="EDIT NOTE"
        onCancel={onCancel}
        onApply={onFormFinish}
        inputValue={titleTodo}
        onChangeInput={onChangeTitleTodo}
      />
    </>
  );
};
