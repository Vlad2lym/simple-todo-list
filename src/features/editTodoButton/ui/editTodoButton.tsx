import styles from '../styles/editTodoButton.module.scss';

interface IProps {
  idTodo: string;
  onClick: (id: string) => void;
}

export const EditTodoButton = ({ idTodo, onClick }: IProps) => {
  return (
    <>
      <button className={styles.editBtn} onClick={() => onClick(idTodo)} />
    </>
  );
};
