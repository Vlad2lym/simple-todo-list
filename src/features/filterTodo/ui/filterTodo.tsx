import { CustomSelect } from '@/shared/ui/customSelect';
import { FilterOption, filterOptions } from '../api/filterTodo';

interface IProps {
  className: string;
  onChangeFilter: (option: FilterOption) => void;
}

export const FilterTodo = ({ className, onChangeFilter }: IProps) => {
  return (
    <>
      <CustomSelect className={className} options={filterOptions} onChange={onChangeFilter} />
    </>
  );
};
