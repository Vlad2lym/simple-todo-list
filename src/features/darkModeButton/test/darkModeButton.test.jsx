import { render, screen } from '@testing-library/react';
import { DarkModeButton } from '../ui/darkModeButton';
import { jest } from '@jest/globals';
import userEvent from '@testing-library/user-event';

const onChange = jest.fn();

it('should change app mode on click', async () => {
  const { container } = render(<DarkModeButton onChange={onChange} />);
  expect(container.firstChild).toMatchSnapshot();
  expect(onChange).not.toHaveBeenCalled();
  await userEvent.click(screen.getByRole('button'));
  expect(onChange).toHaveBeenCalledTimes(1);
});
