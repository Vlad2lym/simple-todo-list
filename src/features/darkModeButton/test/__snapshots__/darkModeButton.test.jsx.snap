// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change app mode on click 1`] = `
<button
  class="darkModeButton"
/>
`;
